File: algorithm/力扣热题100/5_最长回文子串.js
var longestPalindrome = function (s) {
    let res = ''
    for (let i = 0; i < s.length; i++) {
        let str = palindrome(s, i, i)
        if (res.length < str.length) res = str
        str = palindrome(s, i, i + 1)
        if (res.length < str.length) res = str
    }
    return res
}
function palindrome(s, i, j) {
    while (i >= 0 && j < s.length && s[i] === s[j]) {
        i--; j++;
    }
    return s.slice(i + 1, j)
}

File: algorithm/力扣热题100/5_最长回文子串.go
package main

func longestPalindrome(s string) string {
	res := ""
	for i := 0; i < len(s); i++ {
		str := palindrome(s, i, i)
		if len(str) > len(res) {
			res = str
		}
		str = palindrome(s, i, i+1)
		if len(str) > len(res) {
			res = str
		}
	}
	return res
}

func palindrome(s string, i int, j int) string {
	for i >= 0 && j < len(s) && s[i] == s[j] {
		i--
		j++
	}
	return s[i+1 : j]
}
